components/core/src/clp/NetworkReader.cpp

```diff
@@ -6,14 +6,18 @@
 #include <memory>
 #include <mutex>
 #include <optional>
+#include <string>
 #include <string_view>
+#include <unordered_map>
+#include <utility>
 
 #include <curl/curl.h>
 
 #include "CurlDownloadHandler.hpp"
 #include "CurlOperationFailed.hpp"
 #include "ErrorCode.hpp"
 #include "Platform.hpp"
+#include "TraceableException.hpp"
```

Comment:
Do we need this?

Reply:
clang-tidy told me yes, but without it we can also pass the compilation

Reply:
shall I get rid of it? This is the header file for __FILENAME__:
 throw OperationFailed(ErrorCode_Failure, __FILENAME__, __LINE__);

Reply:
Thinking about it again. I shall get rid of it, becauase __FILENAME__ is not part of my code in this PR